interpreter/cling/lib/Interpreter/NullDerefProtectionTransformer.cpp

```diff
@@ -81,10 +86,48 @@ namespace cling {
           if (ArgIndexs.test(index)) {
             // Get the argument with the nonnull attribute.
             Expr* Arg = CE->getArg(index);
-            CE->setArg(index, SynthesizeCheck(Arg));
+            if (Arg->getType().getTypePtr()->isPointerType())
+              CE->setArg(index, SynthesizeCheck(Arg));
           }
         }
       }
+      return true;
+    }
+
+    bool VisitFunctionDecl(FunctionDecl* FD) {
+      if (!FD->isConstexpr()) {
+        CompoundStmt* CS = dyn_cast_or_null<CompoundStmt>(FD->getBody());
+        if (CS)
+          VisitStmt(CS);
+      }
+      return true;
+    }
+
+    bool VisitCXXMethodDecl(CXXMethodDecl* FD) {
```

Comment:
I just checked in gdb, both TraverseX and VisitX are used because of the recursion in the visitor, it first hits Traverse and if I don't reimplement traverse it will use the default one